test/jdk/java/util/IdentityHashMap/Basic.java

```diff
@@ -0,0 +1,628 @@
+    }
+
+    // putIfAbsent
+    @Test
+    public void testPutIfAbsentNoop() {
+        Box r = map.putIfAbsent(k1a, new Box(v1a)); // no-op
+
+        assertSame(r, v1a);
+        checkEntries(map.entrySet(), entry(k1a, v1a),
+                                     entry(k1b, v1b),
+                                     entry(k2, v2));
+    }
+
+    // putIfAbsent
+    @Test
+    public void testPutIfAbsentAddsNew() {
+        Box newKey = new Box(k1a);
+        Box newVal = new Box(v1a);
+        Box r = map.putIfAbsent(newKey, newVal); // adds new entry
+
+        assertNull(r);
+        checkEntries(map.entrySet(), entry(k1a, v1a),
+                                     entry(k1b, v1b),
+                                     entry(k2, v2),
+                                     entry(newKey, newVal));
+    }
+
+    // remove(Object)
+    @Test
+    public void testRemoveKey() {
+        Box r = map.remove(k1b);
+
+        assertSame(r, v1b);
+        checkEntries(map.entrySet(), entry(k1a, v1a),
+                                     entry(k2, v2));
+    }
+
+    // AN: key absent, remappingFunction returns null
+    @Test
+    public void testComputeAN() {
+        Box newKey = new Box(k1a);
+        Box r = map.compute(newKey, (k, v) -> null);
+
+        assertNull(r);
+        checkEntries(map.entrySet(), entry(k1a, v1a),
+                                     entry(k1b, v1b),
+                                     entry(k2, v2));
+    }
+
+    // AV: key absent, remappingFunction returns non-null value
+    @Test
+    public void testComputeAV() {
+        Box newKey = new Box(k1a);
+        Box newVal = new Box(v1a);
+        Box r = map.compute(newKey, (k, v) -> newVal);
+
+        assertSame(r, newVal);
+        checkEntries(map.entrySet(), entry(k1a, v1a),
+                                     entry(k1b, v1b),
+                                     entry(k2, v2),
+                                     entry(newKey, newVal));
+    }
+
+    // PN: key present, remappingFunction returns null
+    @Test
+    public void testComputePN() {
+        Box r = map.compute(k1a, (k, v) -> null);
+
+        assertNull(r);
+        checkEntries(map.entrySet(), entry(k1b, v1b),
+                                     entry(k2, v2));
+    }
+
+    // PV: key present, remappingFunction returns non-null value
+    @Test
+    public void testComputePV() {
+        Box newVal = new Box(v1a);
+        Box r = map.compute(k1a, (k, v) -> newVal);
+
+        assertSame(r, newVal);
+        checkEntries(map.entrySet(), entry(k1a, newVal),
+                                     entry(k1b, v1b),
+                                     entry(k2, v2));
+    }
+
+    @Test
+    public void testComputeIfAbsentIsCalled() {
+        boolean[] called = new boolean[1];
+        Box newKey = new Box(k1a);
+        Box newVal = new Box(v1a);
+        Box r = map.computeIfAbsent(newKey, k -> { called[0] = true; return newVal; });
```

Comment:
More of a curiosity than a review comment - I see that various places in this PR use a boolean array with one element instead of just a boolean. Is that a personal coding preference or is there something more to it?

Reply:
Yes, that's it; you can't mutate a local variable captured by a lambda.

Reply:
Thank you @liach and Stuart. I had overlooked the lambda aspect of this.